util/awsservice/cloudwatchmetrics.go

```diff
@@ -134,6 +134,78 @@ func GetMetricStatistics(
 	return CwmClient.GetMetricStatistics(ctx, &metricStatsInput)
 }
 
+func GetMetricMaximum(
```

Comment:
Curious to hear the reasoning behind this func...is it because we don't know the exact value of the metric? But we expect the value to never be above a certain amount?

Reply:
The reasoning behind this function was to get the highest metric value being emitted for `tomcat.sessions` / `tomcat.rejected_sessions` in order to check if that value is greater than zero since we need to verify we're able to get these metrics above zero. The reason I went for the maximum was just to be sure we check for a value that is absolutely expected to be above zero. 

However, thinking through it, a more efficient solution would be to dedicate the function to check if it's above zero directly and exit out when we find that. I fixed the function to account for that.

As for the exact values for each metric, `tomcat.sessions` will max out to 2 since that's what I set it to in the application, but I didn't think it was necessary to directly check for this since the agent doesn't generate the metric, the application does. But it is important to make sure the agent is able to ship a non-zero value to CloudWatch. Another reason is a possible race condition where there are latency issues and it doesn't append to the right value in time compared to when we check the metric value, so it just seems more ideal to check if the value is above 0 instead of a specific value. `tomcat.rejected_sessions` is unpredictable since there isn't a max value for that, so it'll just keep increasing after the max sessions are hit.